File: Python/Warmup-2/array123.py
"""
Dado um Array "nums", retorne true se nele houver a sequência: 1, 2, 3. Em qualquer
lugar.
"""


def array123(nums):
    for i in range(len(nums)-2):
        if nums[i:i+3] == [1, 2, 3]:
            return True
    return False


print(array123([4, 1, 2, 3]))

File: Java/Warmup-2/array123/Array123.java
package array123;

import java.util.Arrays;

public class Array123 {
    //Retorne true se a sequência "1,2,3" aparecer em qualquer canto do Array.

    public boolean array123(int[] nums) {
        if (nums.length < 3) return false;
        for (int i = 0; i < nums.length-2; i++) {
            if (Arrays.equals(new int[]{nums[i], nums[i + 1], nums[i + 2]}, new int[]{1, 2, 3})){
                return true;
            }
        }
        return false;
    }
}
